pkg/cluster/hive_test.go

```diff
@@ -272,3 +272,51 @@ func createManagerForTests(t *testing.T, existingNamespaceName string) *manager
 	}
 	return m
 }
+
+func TestHiveEnsureResources(t *testing.T) {
+	ctx := context.Background()
+	for _, tt := range []struct {
+		testName           string
+		clusterManagerMock func(mockCtrl *gomock.Controller, m *manager) *mock_hive.MockClusterManager
+		wantErr            string
+	}{
+		{
+			testName: "returns error if cluster manager returns error",
+			clusterManagerMock: func(mockCtrl *gomock.Controller, m *manager) *mock_hive.MockClusterManager {
+				mockClusterManager := mock_hive.NewMockClusterManager(mockCtrl)
+				mockClusterManager.EXPECT().CreateOrUpdate(gomock.Any(), gomock.Any(), gomock.Any()).Return(fmt.Errorf("cluster manager error"))
```

Comment:
```suggestion
				mockClusterManager.EXPECT().CreateOrUpdate(ctx, m.subscriptionDoc, m.doc).Return(fmt.Errorf("cluster manager error"))
```

As in "We expect `CreateOrUpdate` to be called with arguments X, Y and Z, but we want to return an error"

In another case it will be "We expect `CreateOrUpdate` to be called with arguments X, Y and Z, but we want to return no error"

Reply:
I think each test should only test one functionality. The purpose of this unit test is to determine that the behavior of the function is correct in case of an error, regardless of the parameters used, while the other test takes care of the parameters.

If we were to test parameters in both tests, and we introduce a change that breaks the correctness of the parameters being passed to the Cluster Manager, it would not be so obvious that we broke, as both tests would fail.

Reply:
I agree with Mikalai here. For this error, we want to be as specific as possible. This lets us know the exact intent of the code we are testing, which is more important than coverage/just having the test at all.